### ISSUE TITLE
Problem with importing `Peripherals` from crate root

### ISSUE THREAD

**Issue by katyo** [adamgreig/stm32ral#16]
If I understand correct when device feature is used ("stm32f103" in my case) the types from corresponding submodule (i.e. from `stm32f1::stm32f103`) is available from crate's root.
But seems something went wrong in my case:
```
error[E0432]: unresolved import `stm32ral::Peripherals`
 --> src/init.rs:1:65
  |
1 | use stm32ral::{modify_reg, rcc, read_reg, reset_reg, write_reg, Peripherals};
```

My manifest looks like:
```toml
[dependencies.stm32ral]
version = "^0.5"
features = ["rtic", "stm32f103"]
```
```
$ rustc --version
rustc 1.50.0 (cb75ad5db 2021-02-10)
```

I generated doc to see the re-exports. It looks as expected:
```
Re-exports
pub use cortex_m::armv7m::*;
pub use stm32f1::stm32f103::*;
```

Import from `stm32f1::stm32f103` works fine.

**Comment by adamgreig** [comment]
Hmm, weird. I'm not sure why `Peripherals` isn't being re-exported properly. When I've used stm32ral with rtic, I have:

```rust
#[rtic::app(device=stm32ral::stm32f1::stm32f103, peripherals=true)]
```

which seems to work fine, and I don't touch the `Peripherals` struct myself at all. I'll try and figure out where Peripherals is going but it's not obvious at first glance...

Incidentally, you almost certainly want the `rt` feature enabled as well.

**Comment by katyo** [comment]
@adamgreig Of course, thanks :+1:

**Comment by adamgreig** [comment]
I figured out the cause of this issue; both `stm32ral::cortex_m::arm7m` and `stm32ral::stm32f1::stm32f103` (for example) export a `Peripherals` type, so the glob import somehow ends up importing neither and just says `Peripherals` isn't available. Inside a single module it detects this ambiguity and throws an error, but when it's an external crate it seems it doesn't know how to tell you about it. I didn't have much luck finding an actual reference for this behaviour.

Ultimately this probably needs a different generation of the cortex-m shared peripherals (e.g. how some things name them CorePeripherals and DevicePeripherals) but for now I think your best bet is to refer to it by the full path or use it from `stm32ral::stm32f1::stm32f103` as you've done.

**Comment by katyo** [comment]
How about re-exporting things via submodules like a `stm32ral::core::Peripherals` and `stm32ral::device::Peripherals`?

**Comment by adamgreig** [comment]
This is now fixed in v0.8.0 by renaming the cortex-m peripherals to CorePeripherals, thanks for the report!
